LibreNMS/Snmptrap/Handlers/AdvaAccThresholdCrossingAlert.php

```diff
@@ -43,11 +44,97 @@ class AdvaAccThresholdCrossingAlert implements SnmptrapHandler
      */
     public function handle(Device $device, Trap $trap)
     {
-        $raw = $trap->getRaw();
-        Log::event($raw, $device->device_id, 'trap', 2);
         $interval = $trap->getOidData($trap->findOid("CM-PERFORMANCE-MIB::cmEthernetAccPortThresholdInterval"));
         $ifName = $trap->getOidData($trap->findOid("IF-MIB::ifName"));
+        $thresholdMessage = $this->handleThreshold($trap);
+        Log::event("$ifName $thresholdMessage threshold exceeded for $interval", $device->device_id, 'trap', 2);
+    }
+    public static function handleThreshold($trap)
+    {
         $threshOid = $trap->getOidData($trap->findOid("CM-PERFORMANCE-MIB::cmEthernetAccPortThresholdVariable"));
-        Log::event("$ifName threshold exceeded for $interval. Threshold OID is $threshOid.", $device->device_id, 'trap', 2);
+        switch (true) {
+            case strpos($threshOid, 'CM-PERFORMANCE-MIB::cmEthernetAccPortStatsUAS') === 0:
```

Comment:
FYI, we have a function called str_contains() it is a little nicer than strpos :)

Reply:
Made requested changes.

Reply:
Not sure why the changes you made are causing errors. Functionally it all looks the same, just a few changes to where things are called.

Reply:
I this is some kind of scope issue that I don't understand (new to php). If I call getThresholds from withing the handle() method and pass it to getThresholdMessage method it works. Would that be acceptable?

Reply:
Found and fixed the issue.